### PR TITLE
Improve function call matching for intrinsics

### PR DESCRIPTION
**Pull request by ehsantn**
Intrinsics were sometimes found as being from `numba.extending` module, which is fixed by this patch.

### CODE REVIEW

File: numba/tests/test_parfors.py [hunk 1]
```diff
@@ -1142,6 +1144,19 @@ def test_impl(A, B):
         B[1] = 0
         self.check(test_impl, A, B)
 
+    @skip_unsupported
+    def test_find_callname_intrinsic(self):
+        from numba.unsafe.ndarray import empty_inferred as unsafe_empty
+        def test_impl(n):
+            A = unsafe_empty((n,))
+            for i in range(n):
+                A[i] = 2.0+i
```
Comment (stuartarchibald) [inline numba/tests/test_parfors.py]: spaces around +, PEP8.

Comment (ehsantn) [reply numba/tests/test_parfors.py]: done.

**Comment by stuartarchibald** [comment]
Thanks for the fix up, once CI passes this can be merged.

**Comment by seibert** [comment]
CI passed.  Merging.

**Pull request merged**